### PR TITLE
Fix for -c and -p in merge and sort

### PR DESCRIPTION
**Pull request by mp15**
-Add fix so that -c and -p put the first @RG or @PG into the output
-Add testcase to check the -c and -p

### CODE REVIEW

**Comment by mp15** [comment]
Fixes #411

File: bam_sort.c [hunk 1]
```diff
@@ -307,14 +307,15 @@ static void trans_tbl_init(bam_hdr_t* out, bam_hdr_t* translate, trans_tbl_t* tb
         regcomp(&rg_id_search, rg_regex.s, REG_EXTENDED|REG_NEWLINE|REG_NOSUB);
         free(rg_regex.s);
         kstring_t transformed_id = { 0, 0, NULL };
-        bool transformed_equals_match;
+        bool transformed_equals_match, not_found_in_output;
```
Comment (jkbonfield) [inline bam_sort.c]: I'm a bit unsure of how not_found_in_output is initialised.  It only gets set when rg_override is NULL.  Maybe we need a default value?

I couldn't see how to produce a test case where rg_override is NULL though so I don't know if this is possible.

**Comment by jkbonfield** [comment]
Other than the one minor point, I have no other review comments to make.

**Comment by mp15** [comment]
Fix #416

**Comment by jmarshall** [comment]
Merged as 227c142c288b4757e9bd7907a0b52a9f16ee451b.

**Pull request closed without merging**